nextcord/ext/commands/bot.py

```diff
@@ -901,6 +902,88 @@ def reload_extension(self, name: str, *, package: Optional[str] = None) -> None:
             # revert sys.modules back to normal and raise back to caller
             sys.modules.update(modules)
             raise
+    
+    def load_extensions(
+            self, 
+            names: List[str], 
+            *, 
+            package: Optional[str] = None,
+    ) -> List[str]:
+        """Loads all extensions provided in a list.
+        
+        Parameters
+        ----------
+        names: :type:`List[str]`
+            The names of all of the extensions to load.
+        package: :type:`Optional[str]`
+            The package name to resolve relative imports with.
+            This is required when loading an extension using a relative path, e.g ``.foo.test``.
+            Defaults to ``None``.
+
+        Returns
+        --------
+        A :type:`List[str]` that contains the names of all of the extensions
+        that loaded successfully.
+
+        Raises
+        --------
+        ExtensionNotFound
+            An extension could not be imported.
+        ExtensionAlreadyLoaded
+            An extension is already loaded.
+        NoEntryPointError
+            An extension does not have a setup function.
+        ExtensionFailed
+            An extension or its setup function had an execution error.
+        """
+        # TODO: Extra arguments
+
+        loaded_extensions: List[str] = []
+
+        for i_extension in range(len(names)):
+            try:
+                self.load_extension(names[i_extension], package=package)
+            except Exception:
+                raise
+            else:
+                loaded_extensions.append(names[i_extension])
+
+        return loaded_extensions
+
+    def find_load_extensions(self, folder_name: str) -> List[str]:
+        """Loads all extensions found in a folder.
+
+        Once an extension found in a folder has been loaded and did not throw
+        any exceptions, it will be added to a list of extension names that
+        will be returned.
+
+        Parameters
+        ----------
+        folder_name: :class:`str`
+            The name (or path) of the folder to look through.
+
+        Returns
+        --------
+        A :type:`List[str]` that contains the names of all of the extensions
+        that loaded successfully.
+
+        Raises
+        --------
+        ExtensionNotFound
+            An extension could not be imported.
+        ExtensionAlreadyLoaded
+            An extension is already loaded.
+        NoEntryPointError
+            An extension does not have a setup function.
+        ExtensionFailed
+            An extension or its setup function had an execution error.
+        """
+        # TODO: Extra arguments
+
+        files: List[str] = [f[:-3] for f in os.listdir(folder_name) if f.endswith(".py")]
```

Comment:
I dislike this, what happens when people have an ``__init__.py`` file? What happens if people have nested cog setups (I see it somewhat regularly?

Reply:
Someone else pointed out the nested cog setups, and I said I'll try to work out some recursive solution.
Also, if the file is not for a cog, then there isn't really a good way to know unless we try to load it